### ISSUE TITLE
The heartbeat call to ping() sometimes occurs after the transport is closing

### ISSUE THREAD

**Issue by WillCodeCo** [aio-libs/aiohttp#5766]
🐞 **Describe the bug**

When a websocket is setup with a `heartbeat` the `ClientWebSocketResponse._send_heartbeat()` function is periodically called, which creates a fire-and-forget task in order to call `self._writer.ping()`.

The problem is that this task does not check to see if the transport is closed/closing before calling `_writer.ping()`

```python
    def _send_heartbeat(self) -> None:
        if self._heartbeat is not None and not self._closed:
            # fire-and-forget a task is not perfect but maybe ok for
            # sending ping. Otherwise we need a long-living heartbeat
            # task in the class.
            self._loop.create_task(self._writer.ping())

            if self._pong_response_cb is not None:
                self._pong_response_cb.cancel()
            self._pong_response_cb = call_later(
                self._pong_not_received, self._pong_heartbeat, self._loop
            )
           
```

💡 **To Reproduce**

```python
import aiohttp
from aiohttp import web
import asyncio
import random

class ExampleServer:
    def __init__(self, host: str, port: int):
        self._host = host
        self._port = port
        self._app = web.Application()
        self._app.add_routes([web.get('/ws', self._get_ws_handler)])

    @classmethod
    async def _run_server(cls, web_app, host: str, port: int):
        try:
            runner = web.AppRunner(web_app)
            await runner.setup()
            site = web.TCPSite(runner, host, port)
            await site.start()
            print(f"HTTP Server is starting at {host}:{port}")
            while True:
                await asyncio.sleep(60)
        finally:
            print("HTTP Server is terminating")
            await runner.cleanup()

    async def _get_ws_handler(self, request: web.Request) -> web.Response:
        ws = web.WebSocketResponse(heartbeat=0.1)
        await ws.prepare(request)
        async for received_msg in ws:
            print(f"Server received: {received_msg.data}")

    async def run(self):
        await self._run_server(self._app, self._host, self._port)


async def run_server():
    server = ExampleServer('localhost', 8888)
    await server.run()


async def run_client(session, rng, client_name):
    async with session.ws_connect(url='ws://localhost:8888/ws', heartbeat=0.1) as ws:
        await ws.send_bytes(b'hello from '+client_name.encode('ascii'))
        await asyncio.sleep(rng.randint(100,1000)/1000)
        await ws.close()


async def main():
    rng = random.Random(42)
    exceptions_raised = []
    asyncio.get_event_loop().set_exception_handler(lambda loop, context: exceptions_raised.append(context.get('exception')))
    asyncio.create_task(run_server())
    await asyncio.sleep(0.5)
    async with aiohttp.ClientSession() as session:
        for x in range(1000):
            asyncio.create_task(run_client(session, rng, f'client_{x}'))
        await asyncio.sleep(5)
        if exceptions_raised == []:
            print("No exceptions were raised :)")
        else:
            for e in exceptions_raised:
                raise e

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```
Add these steps below this comment: -->

💡 **Expected behavior**

No exceptions should be raised

📋 **Logs/tracebacks**

```python-traceback (paste your traceback in the next line)
Traceback (most recent call last):
  File "/tmp/badping.py", line 70, in <module>
    loop.run_until_complete(main())
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/tmp/badping.py", line 67, in main
    raise e
  File "/tmp/.env/lib/python3.9/site-packages/aiohttp/http_websocket.py", line 673, in ping
    await self._send_frame(message, WSMsgType.PING)
  File "/tmp/.env/lib/python3.9/site-packages/aiohttp/http_websocket.py", line 643, in _send_frame
    self._write(header + mask + message)
  File "/tmp/.env/lib/python3.9/site-packages/aiohttp/http_websocket.py", line 660, in _write
    raise ConnectionResetError("Cannot write to closing transport")
ConnectionResetError: Cannot write to closing transport

```

📋 **Your version of the Python**

```console
$ python --version
Python 3.9.3
```

📋 **Your version of the aiohttp/yarl/multidict distributions**

```console
$ python -m pip show aiohttp
Name: aiohttp
Version: 3.7.4.post0
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: Nikolay Kim
Author-email: fafhrd91@gmail.com
License: Apache 2
Location: /tmp/.env/lib/python3.9/site-packages
Requires: chardet, yarl, typing-extensions, multidict, async-timeout, attrs
Required-by: 

```
```console
$ python -m pip show multidict
Name: multidict
Version: 5.1.0
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /tmp/.env/lib/python3.9/site-packages
Requires: 
Required-by: yarl, aiohttp

```
```console
$ python -m pip show yarl
Name: yarl
Version: 1.6.3
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /tmp/.env/lib/python3.9/site-packages
Requires: multidict, idna
Required-by: aiohttp

```

📋 **Additional context**

Running on 5.4.112-1-MANJARO

**Comment by greshilov** [comment]
Probably a duplicate of #5182

**Comment by WillCodeCo** [comment]
Yeah looks to be a duplicate
